src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/NuGetRestorer.cs

```diff
@@ -134,9 +136,15 @@ private Task ProjectPropertyChangedAsync(Tuple<ImmutableList<IProjectValueVersio
             if (projectRestoreInfo != null)
             {
                 _projectVsServices.Project.Services.ProjectAsynchronousTasks
-                    .RegisterCriticalAsyncTask(JoinableFactory.RunAsync(() => _solutionRestoreService
-                            .NominateProjectAsync(_projectVsServices.Project.FullPath, projectRestoreInfo, CancellationToken.None)),
-                            registerFaultHandler: true);
+                    .RegisterCriticalAsyncTask(JoinableFactory.RunAsync(async () =>
+                    {
+                        await _solutionRestoreService
+                               .NominateProjectAsync(_projectVsServices.Project.FullPath, projectRestoreInfo, CancellationToken.None)
+                               .ConfigureAwait(true);
```

Comment:
This was a late change to prevent deadlocks. Multiple nominate calls while a restore is happening get queued and will return the same task, which completes after the next restore happens.

Still experimenting with the perf team on this change. Right now they are not detecting the marker. Does it need to be called in specific thread, or is a flag needed to activate it?